tests/client/components/pagination.spec.js

```diff
@@ -0,0 +1,177 @@
+  let carousel;
+
+  beforeEach(() => {
+    carousel = {
+      pageCount: 1,
+      paginationPadding: 1,
+      paginationSize: 1,
+      paginationActiveColor: 'red',
+      paginationColor: 'blue',
+      paginationPosition: 'top',
+      slideCount: 2,
+      scrollPerPage: true,
+      currentPerPage: 2,
+      currentPage: 0,
+      $children: [
+        {
+          title: 'title1'
+        }
+      ]
+    };
+  });
+
+  it('should match the stored snapshot', () => {
+    const wrapper = shallowMount(Pagination, { provide: { carousel } });
+
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should not contain a button', () => {
+    carousel.pageCount = 0;
+
+    const wrapper = shallowMount(Pagination, { provide: { carousel } });
+
+    expect(wrapper.find('button').exists()).toBe(false);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should contain a button', () => {
+    const wrapper = shallowMount(Pagination, { provide: { carousel } });
+
+    expect(wrapper.find('button').exists()).toBe(true);
+  });
+
+  it('should set the correct title', () => {
+    const wrapper = shallowMount(Pagination, { provide: { carousel } });
+
+    expect(wrapper.vm.getDotTitle(0)).toBe('title1');
+  });
+
+  it('should emit paginationclick button event with index 0 when button is clicked', () => {
+    const wrapper = shallowMount(Pagination, { provide: { carousel } });
+
+    wrapper.find('button').trigger('click');
+
+    expect(wrapper.emitted().paginationclick[0][0]).toBe(0);
+  });
+
+  describe('paginationCount', () => {
+    it('should be set to pageCount if scrollPerPage is enabled', () => {
+      carousel.scrollPerPage = true;
+      carousel.pageCount = 1;
+
+      const wrapper = shallowMount(Pagination, { provide: { carousel } });
+
+      expect(wrapper.vm.paginationCount).toBe(1);
+    });
+
+    it('should be set to 0 if the currentPerPage is 0 and scrollPerPage is disabled', () => {
+      carousel.scrollPerPage = false;
+      carousel.currentPerPage = 0;
+
+      const wrapper = shallowMount(Pagination, { provide: { carousel } });
+
+      expect(wrapper.vm.paginationCount).toBe(0);
+    });
+
+    it('should be set to 1 if the currentPerPage is 2, slideCount is 2 and scrollPerPage is disabled', () => {
+      carousel.scrollPerPage = false;
+      carousel.currentPerPage = 2;
+      carousel.slideCount = 2;
+
+      const wrapper = shallowMount(Pagination, { provide: { carousel } });
+
+      expect(wrapper.vm.paginationCount).toBe(1);
+    })
+  });
+
+  describe('isCurrentDot', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      wrapper = shallowMount(Pagination, { provide: { carousel } });
+    });
+
+    it('should be false when not the current dot', () => {
+      expect(wrapper.vm.isCurrentDot(1)).toBe(false);
+    });
+
+    it('should be true when it is the current dot', () => {
+      expect(wrapper.vm.isCurrentDot(0)).toBe(true);
+    })
+  });
+
+  // TODO: What is with this expected behaviour?
```

Comment:
This is the current logic in the component, it really doesn't make sense so leaving this todo here so we can address in the future